pkg/trace/agent/agent.go

```diff
@@ -561,8 +561,15 @@ func (a *Agent) ProcessStats(in *pb.ClientStatsPayload, lang, tracerVersion stri
 	a.ClientStatsAggregator.In <- a.processStats(in, lang, tracerVersion)
 }
 
-// sample performs all sampling on the processedTrace modifying it as needed and returning if the trace should be kept and the number of events in the trace
+// sample performs all sampling on the processedTrace modifying it as needed and returning if the trace should be kept
+// and the number of events in the trace
 func (a *Agent) sample(now time.Time, ts *info.TagStats, pt *traceutil.ProcessedTrace) (keep bool, numEvents int) {
+	// If the agent is set for Error Tracking Standalone only the ErrorSampler is run (bypasses all other samplers).
+	// Trace chunks that don't contain errors are dropped.
+	if a.conf.ErrorTrackingStandalone {
+		return a.errorSampling(now, ts, pt)
+	}
```

Comment:
That may be more of a Product question but here is the reasoning:

- if a user sets a host to ETS, we only want to keep chunks with errors (or exception span events)
- the error sampler itself may not be needed - likely users would like all their errors and then use exclusion filters - but we can still leave it if they want to sample
- apart from the priority sampler with manual drop, all sampling decisions could be to keep regardless of the presence of errors - which we don't want as we only need to send chunks with errors

Then I have to admit I didn't know the probabilistic sampler talked to the tracer, but I guess we wouldn't need that if the host is set as ETS and we never run it?

I discussed with @dussault-antoine this week and we concluded that it was fine to bypass all samplers but the error sampler.

Then I don't remember if I considered putting the logic in `runSamplers()`, maybe that would give better readability?

Reply:
👍 From a product perspective, thanks for confirming that.

For readability and to keep things in the same place, I believe `runSamplers` will be a better place to decide which samplers to run, yes.

Reply:
Unfortunately I think I need to answer this question with another question: If ErrorTrackingStandalone is disabled does that mean the customer is for sure NOT using APM and should not be able to view complete traces?
If so we should document this config value more explicitly with that in mind. There are a lot of features / functionality in the trace-agent so I worry that having this sort of "special mode" might accidentally infect a bunch of other features of the trace-agent (or that some thing will be missed and customers might accidentally enable or be billed for products unexpectedly) Is there an RFC or design doc I can review here?

Reply:
Is it ok if it goes through SingleSpanSampling? if we don't keep a chunk, hopefully it is not kept

Reply:
@ajgajg1134 ETS disabled is standard APM
ETS enabled acts as a filter on APM: the agent only sends chunks that contain errors

Here is the [product brief](https://docs.google.com/document/d/13ITxGJal0WO1zxjr3YDEdJkAGmlTmwam23cLFV3QhhA/edit?tab=t.0#heading=h.9hjfhrytnmop), and [another doc ](https://docs.google.com/document/d/1PoKs0D3CER3rL7jDlMEQGxkgMMKzkKibB44ygu6xFmc/edit?tab=t.0#heading=h.n0avhkm5m873) from @meghanlo

Reply:
Moved sampling logic in `runSamplers()` in https://github.com/DataDog/datadog-agent/pull/30065/commits/96d9668e722c806ed325dc2ae8f6950e363eae39.

Also added a check in `sample()` to avoid single span sampling.